pixelmod/src/main/java/io/pixelmod/repository/DecalHandle.kt

```diff
@@ -0,0 +1,13 @@
+package io.pixelmod.repository
+
+import android.graphics.Bitmap
+
+/**
+ * A pointer to one of the items in the IDecalRepository.
+ */
+data class DecalHandle(
```

Comment:
can any of these fields be nullable?

Reply:
No

Reply:
It actually depends on you. These information can be displayed in the UI. id for sure isn't optional. Other fields may be null if this doesn't complicate things.

Reply:
Can we make the handler not to store anything besides the Bitmap? simply pass in a Bitmap and returns a Bitmap? 
The handler should not know description or title, it should only swap the logo.

Reply:
This isn't a handler function. A handle is just like a pointer. In this case, it points to one of the items in a decal repository. The handle contains generic information attached to a decal.

The App can select a preferred repository but doesn't have prior knowledge of what decals are contained in a repository. The app can query the repository to see the set of available decals.

Information about the collection of available repositories is app-related data, and the app is responsible for deciding which one to use. However, a specific repository is pixelmod related data because only pixelmod can understand the structure/format of the repository. This is particularly true when it uses a model-based repository. The set of decals in this kind of repository is strictly predefined by pixelmod.

Reply:
ok. LGTM!